### PR TITLE
iOS 11 support

### PR DESCRIPTION
**Pull request by brandons**
- Includes scroll view support from https://github.com/MoZhouqi/KMNavigationBarTransition/pull/81
- Fixes navigation bar height on iOS 11 https://stackoverflow.com/a/45331362/324464

### CODE REVIEW

**Comment by brandons** [comment]
@MoZhouqi Can you review and merge?

File: KMNavigationBarTransition/UINavigationBar+KMNavigationBarTransition.m [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+}
+
+- (BOOL)km_useExtendedHeight {
+	return [objc_getAssociatedObject(self, _cmd) boolValue];
+}
+
+- (void)setKm_useExtendedHeight:(BOOL)useHeight {
+	objc_setAssociatedObject(self, @selector(km_useExtendedHeight), @(useHeight), OBJC_ASSOCIATION_ASSIGN);
+	[self setNeedsLayout];
+	[self layoutIfNeeded];
```
Comment (MoZhouqi) [inline KMNavigationBarTransition/UINavigationBar+KMNavigationBarTransition.m]: This line of code may be unnecessary.

File: KMNavigationBarTransition/UIViewController+KMNavigationBarTransition.m [hunk 2]
```diff
@@ -118,18 +122,40 @@ - (void)km_addTransitionNavigationBarIfNeeded {
...
 - (void)km_adjustScrollViewContentOffsetIfNeeded {
     if ([self.view isKindOfClass:[UIScrollView class]]) {
-        UIScrollView *scrollView = (UIScrollView *)self.view;
-        const CGFloat topContentOffsetY = -scrollView.contentInset.top;
-        const CGFloat bottomContentOffsetY = scrollView.contentSize.height - (CGRectGetHeight(scrollView.bounds) - scrollView.contentInset.bottom);
-        
-        CGPoint adjustedContentOffset = scrollView.contentOffset;
-        if (adjustedContentOffset.y > bottomContentOffsetY) {
-            adjustedContentOffset.y = bottomContentOffsetY;
+#ifdef __IPHONE_11_0
```
Comment (MoZhouqi) [inline KMNavigationBarTransition/UIViewController+KMNavigationBarTransition.m]: The code logic is duplicated, so the following code may be better.
```
UIEdgeInsets contentInset;
if (@available(iOS 11.0, *)) {
    contentInset = scrollView.adjustedContentInset;
} else {
    contentInset = scrollView.contentInset;
}
```

Comment (MoZhouqi) [inline KMNavigationBarTransition/UIViewController+KMNavigationBarTransition.m]: The code logic is duplicated, so the following code may be better.
```objc
        UIEdgeInsets contentInset;
#ifndef __IPHONE_11_0
        if (@available(iOS 11.0, *)) {
            contentInset = scrollView.adjustedContentInset;
        }
#else
        contentInset = scrollView.contentInset;
#endif
```

**Comment by MoZhouqi** [comment]
Thanks @brandons, The pull request has been involved in a360bacfb7674daa041ac057514bb95634e11254.

**Pull request closed without merging**